platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/gossip/shadowgraph/Shadowgraph.java

```diff
@@ -453,12 +456,28 @@ private void expire(final ShadowEvent shadow) {
      * @param e The event.
      * @return the shadow event that references an event, or null is {@code e} is null
      */
-    public synchronized ShadowEvent shadow(final EventImpl e) {
+    @Nullable
+    public synchronized ShadowEvent shadow(@NonNull final List<EventDescriptorWrapper> e) {
```

Comment:
I don't think its great for an a method to have a list as input and then ignore all but the first element. I assume this is for other parents. Maybe it should be stated that this is for this case and maybe throw if an event has more than one other parent

Reply:
done